deploy/00_register_other.ts

```diff
@@ -0,0 +1,156 @@
+      addr,
+      {
+        value: price,
+      },
+    )
+    console.log(`Registering name ${label}.eth (tx: ${registerTx.hash})...`)
+    await registerTx.wait()
+
+    if (records) {
+      const _publicResolver = publicResolver.connect(await ethers.getSigner(registrant))
+
+      const hash = namehash(`${label}.eth`)
+      console.log(`Setting records for ${label}.eth...`)
+      if (records.text) {
+        console.log('TEXT')
+        for (const { key, value } of records.text) {
+          const setTextTx = await _publicResolver.setText(hash, key, value)
+          console.log(` - ${key} ${value} (tx: ${setTextTx.hash})...`)
+          await setTextTx.wait()
+        }
+      }
+      if (records.addr) {
+        console.log('ADDR')
+        for (const { key, value } of records.addr) {
+          const setAddrTx = await _publicResolver['setAddr(bytes32,uint256,bytes)'](
+            hash,
+            key,
+            value,
+          )
+          console.log(` - ${key} ${value} (tx: ${setAddrTx.hash})...`)
+          await setAddrTx.wait()
+        }
+      }
+      if (records.contenthash) {
+        console.log('CONTENTHASH')
+        const setContenthashTx = await _publicResolver.setContenthash(hash, records.contenthash)
+        console.log(` - ${records.contenthash} (tx: ${setContenthashTx.hash})...`)
+        await setContenthashTx.wait()
+      }
+
+      if (migrate) {
+        const latestResolver = await ethers.getContract('PublicResolver')
+        const _latestResolver = latestResolver.connect(await ethers.getSigner(registrant))
+
+        console.log(`Migrating records for ${label}.eth...`)
+        if (records.text) {
+          console.log('TEXT')
+          for (const { key, value } of records.text) {
+            const setTextTx = await _latestResolver.setText(hash, key, value)
+            console.log(` - ${key} ${value} (tx: ${setTextTx.hash})...`)
+            await setTextTx.wait()
+          }
+        }
+        if (records.addr) {
+          console.log('ADDR')
+          for (const { key, value } of records.addr) {
+            const setAddrTx = await _latestResolver['setAddr(bytes32,uint256,bytes)'](
+              hash,
+              key,
+              value,
+            )
+            console.log(` - ${key} ${value} (tx: ${setAddrTx.hash})...`)
+            await setAddrTx.wait()
+          }
+        }
+        if (records.contenthash) {
+          console.log('CONTENTHASH')
+          const setContenthashTx = await _latestResolver.setContenthash(hash, records.contenthash)
+          console.log(` - ${records.contenthash} (tx: ${setContenthashTx.hash})...`)
+          await setContenthashTx.wait()
+        }
+      }
+    }
+  }
+
+  await network.provider.send('anvil_setBlockTimestampInterval', [1])
+
+  return true
+}
+
+func.id = 'register-other-names'
+func.tags = ['register-other-names']
+func.dependencies = ['LegacyETHRegistrarController']
+func.runAtTheEnd = true
+
+export default func
```

Comment:
this is for normal legacy registrations as well, you don't need the other deploy file if youre using this one

Reply:
I originally merged the two files and then I undid it with some name changes to try to be clearer.

The reason for the unmerge is that the runtime for 00_register_legacy is hard-coded in ensdomains/contracts and at that time the PublicResolver contract will not have been deployed and will throw an error. I think we can consider refactoring the tagging system and such so that contracts are all deployed first and then the data is added.

Reply:
oh right i see now, in this case you can just keep the normal registration and record setting in the original deploy file, then create a new deploy file that specifically sets records on the new resolver. doing that makes sure there's less duplication of code